django/contrib/admin/widgets.py

```diff
@@ -432,8 +434,9 @@ def optgroups(self, name, value, attr=None):
         if not self.is_required and not self.allow_multiple_selected:
             default[1].append(self.create_option(name, '', '', False, 0))
         choices = (
-            (obj.pk, self.choices.field.label_from_instance(obj))
-            for obj in self.choices.queryset.using(self.db).filter(pk__in=selected_choices)
+            (getattr(obj, self.to_field) if self.to_field else obj.pk, self.choices.field.label_from_instance(obj))
```

Comment:
Why not set `self.to_field` to `pk` by default, to save you the exception handling here?